src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -629,9 +632,20 @@ export const Deposit: React.FC<DepositProps> = ({
     ],
   )
 
-  const validateFiatAmount = useCallback(
+  const validateCryptoAmount = useCallback(
+    (value: string) => {
+      contextDispatch?.({ type: ThorchainSaversDepositActionType.SET_LOADING, payload: true })
+      return _validateCryptoAmount(value).finally(() => {
+        contextDispatch?.({ type: ThorchainSaversDepositActionType.SET_LOADING, payload: false })
+      })
+    },
+    [_validateCryptoAmount, contextDispatch],
+  )
+
+  const _validateFiatAmount = useCallback(
```

Comment:
do we want to return on `!contextDispatch` here as well? and same stylistic question as above

Reply:
ab5b41412d for the early return on `!dispatch`